sdk/src/builder.rs

```diff
@@ -409,6 +482,56 @@ impl Builder {
         Ok(self)
     }
 
+    /// Adds a single action to the manifest.
+    /// This is a convenience method for adding an action to the `Actions` assertion.
+    /// # Arguments
+    /// * `action` - The action name as a string.
+    /// * `data` - The data for the action as a Serde-serializable type.
+    /// # Returns
+    /// * A mutable reference to the [`Builder`].
+    /// # Errors
+    /// * Returns an [`Error`] if the action is not valid.
+    /// # Example
+    /// ```rust
+    /// use c2pa::Builder;
+    /// use serde_json::json;
+    /// let created_action = json!({
+    ///    "action": "c2pa.placed",
+    ///    "digitalSourceType": "http://c2pa.org/digitalsourcetype/empty"
+    /// });
+    ///
+    /// let mut builder = Builder::new();
+    /// builder.add_action(created_action);
+    /// ```     
+    pub fn add_action<T>(&mut self, action: T) -> Result<&mut Self>
+    where
+        T: Serialize,
+    {
+        // Allow actions to be a Actions struct, or JSON string, or a serde_json::Value.
+        let action_value = serde_json::to_value(action)?;
+        let action: Action = serde_json::from_value(action_value).map_err(Error::JsonError)?;
+
+        // if an actions assertion already exists, we will append to it
+        // if not, we will create a new one
+        let actions = if let Some(pos) = self
+            .definition
+            .assertions
+            .iter()
+            .position(|a| a.label == Actions::LABEL)
+        {
+            // Remove and use the existing actions assertion
+            let assertion_def = self.definition.assertions.remove(pos);
+            assertion_def.to_assertion()?
+        } else {
+            Actions::new()
+        };
+
+        let actions = actions.add_action(action);
+
+        self.add_assertion(Actions::LABEL, &actions)?;
```

Comment:
Is it possible to serialize the `Action` into json, then construct an `AssertionDefinition` and append it?